### PR TITLE
move e2e into its own go module and remove the runtime dependency on controller-runtime

### PR DESCRIPTION
**Pull request by ecordell**
See: https://github.com/authzed/controller-idioms/pull/40 for more details

### CODE REVIEW

File: .github/workflows/build-test.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ on:  # yamllint disable-line rule:truthy
     branches:
       - "*"
 env:
-  GO_VERSION: "~1.19"
+  GO_VERSION: "~1.20"
```
Comment (josephschorr) [inline .github/workflows/build-test.yaml:14]: move to 1.21 now that its out?

File: pkg/controller/client.go [hunk 2]
```diff
@@ -16,7 +16,7 @@ func (c *Controller) PatchStatus(ctx context.Context, patch *v1alpha1.SpiceDBClu
 	}
 	patch.ManagedFields = nil
 	patch.Status.ObservedGeneration = patch.Generation
-	data, err := client.Apply.Data(patch)
+	data, err := json.Marshal(patch)
```
Comment (josephschorr) [inline pkg/controller/client.go:19]: any semantic differences between this and the only `Apply` stuff?

File: pkg/controller/ensure_deployment_test.go [hunk 3]
```diff
@@ -62,7 +62,7 @@ func TestEnsureDeploymentHandler(t *testing.T) {
 			migrationHash: "testtesttesttest",
 			secretHash:    "secret",
 			existingDeployments: []*appsv1.Deployment{{ObjectMeta: metav1.ObjectMeta{Annotations: map[string]string{
-				metadata.SpiceDBConfigKey: "n549h6bh675h699h697h86h5cchdfq",
+				metadata.SpiceDBConfigKey: "n59hb4h5f7h5b7h5dchddh79h67dq",
```
Comment (josephschorr) [inline pkg/controller/ensure_deployment_test.go:65]: Does these keys changing imply that existing operators, when updated, will cause all resources to be re-applied?

File: pkg/controller/client.go (hunk 2 above)
Comment (ecordell) [reply pkg/controller/client.go:19]: nope: https://github.com/kubernetes-sigs/controller-runtime/blob/304027bcbe4b3f6d582180aec5759eb4db3f17fd/pkg/client/patch.go#L212

File: pkg/controller/ensure_deployment_test.go (hunk 3 above)
Comment (ecordell) [reply pkg/controller/ensure_deployment_test.go:65]: It does, I didn't look into why this happened specifically, but we did jump from 1.26 client go to 1.28 so I assume it's something in the deployment fields.

File: .github/workflows/build-test.yaml (hunk 1 above)
Comment (ecordell) [reply .github/workflows/build-test.yaml:14]: we usually wait a little bit right? I figured we'd do spicedb first (I would want to move to the new stdlb slices/maps libs where possible too)

Comment (josephschorr) [reply .github/workflows/build-test.yaml:14]: K

**Review by josephschorr — APPROVED** [review]
LGTM

**Pull request merged**